### PR TITLE
Bug Fixes and Random Event Weight Drops

### PR DESCRIPTION
**Pull request by Super3222**
Fixes https://github.com/yogstation13/yogstation/issues/339.
- Swarmer event weight lowered to 3 (from 7).
- It can only spawn now 45 minutes into the round (from 30).
- Alien Infestation event weight lowered to 3 (from 5).
- Enthrall now has a check for whether the target is already a thrall or not.
- Enthrall works on loyalists again
- Abstract shoes (shoes like golems feet or shadowling chintin feet) will no longer "show up" when footprints are examined

:cl: Super3222
tweak: Swarmer weight chance is lowered from 5 to 2.
tweak: Alien Infestation event chance is lowered from 5 to 3.
tweak: Brand Intelligence event chance is lowered from 5 to 3.
tweak: Meteor Event chance is lowered from 5 to 3.
tweak: Now you're probably wondering what we're getting from all of these tweaks decreasing the chance of your favorite events. The catch is, that all of them now have a growth feature. So if the event doesn't run, for instance the Alien Event, it's weight will increase every hour the round goes on for. Are you up to the challenge?
tweak: Enthrall now works on loyalists again.
tweak: Golem and Shadowling feet no longer show up in footprints.
/:cl:

### CODE REVIEW

**Comment by MacHac** [comment]
> lowering to 1

I appreciate that they weren't uncommon, but that's going to practically remove them from play.  At least leave it at 2-3.

**Comment by GrayRachnid** [comment]
Agreed with Mac 2-3 sounds better. 1 is very VERY VERY uncommon.

**Comment by oisin100** [comment]
Its not as bad as xeno, bump it up to 2.

**Comment by ghost** [comment]
bump both to 2-3 

aliens are good and swarmers are good also

or instead, make the chance start low and increase scaling based on time passed so that towards the end of a round like 2 hours or shit bad shit starts to happen

**Comment by Super3222** [comment]
the longer you stay on the station, the more of a chance chaos is bound to happen
will look into

**Comment by Super3222** [comment]
In the last few commits:
- Brand Intelligence and Meteor weights have dropped.
- All of the events that have had weight drops now have 'growth' enabled to them, so in the instance that it's been an hour or two, the weight of these events will grow.

File: code/controllers/subsystem/events.dm [hunk 1]
```diff
@@ -90,6 +90,12 @@ var/datum/subsystem/events/SSevent
 
 	for(var/datum/round_event_control/E in control)
 		if(!E.canSpawnEvent(players_amt, gamemode))
+			if (E.growth)
+				var/estimatedtime = PRESUMEDHOUR * E.growth
+				if(world.time > estimatedtime)
+					E.weight++
+					E.growth++
+					message_admins("Random event [E.name] could not spawn, so it's weight has grown to [E.weight].")
```
Comment (oisin100) [inline code/controllers/subsystem/events.dm:96]: Will this not output a spam of messages saying ""Random event [E.name] could not spawn, so it's weight has grown to [E.weight]."" For every event with growth?

Comment (Super3222) [inline code/controllers/subsystem/events.dm:96]: Yeah, I thought admins should know when rounds are close to becoming chaotic.
I should also probably also up some the event weights. 2 is pretty low.

File: code/game/objects/effects/decals/Cleanable/humans.dm [hunk 2]
```diff
@@ -178,6 +178,8 @@
 		. += "You recognise the footprints as belonging to:\n"
 		for(var/shoe in shoe_types)
 			var/obj/item/clothing/shoes/S = shoe
+			if(S.flags & ABSTRACT)
+				break
```
Comment (Alek2ander) [inline code/game/objects/effects/decals/Cleanable/humans.dm]: So examining a footprint made with abstract shoes will tell you "You recognise the footprints as belonging to:" without any follow-up lines
A better way would be to not add abstract shoes to the shoe list in the decal in the first place

Comment (Super3222) [inline code/game/objects/effects/decals/Cleanable/humans.dm]: I don't know, I thought it'd be interesting if an abstract foot within one print would make it unreadable. Making the user question whether they should follow it or not.

**Comment by Super3222** [comment]
Yup, I suspected something like this would happen. Give me a second.

**Pull request closed without merging**